core/src/main/scala/blobstore/BasePath.scala

```diff
@@ -28,7 +28,7 @@ object BasePath {
   def fromString(s: String, forceRoot: Boolean): Option[Path] =
     if (s.isEmpty) Some(empty)
     else
-      scala.util.Try(URI.create(s)).toOption.flatMap { uri =>
+      scala.util.Try(URI.create(s.replaceAll(" ", "%20"))).toOption.flatMap { uri =>
```

Comment:
We should probably url encode all path segments in order to handle other special characters as well, I wonder if we break on paths with <> now.